### PR TITLE
Replace YAML::Tiny with YAML::PP

### PR DESCRIPTION
**Pull request by perlpunk**
YAML::Tiny accepts some invalid YAML syntax, so it's not compatible
with other YAML modules (in perl or other languages).

Also fixed some pod.

So now also Aliases/anchors are supported, and the short "flow style" syntax like:
```
foo: [short, list of, things]
```

It would also possible to support `<<` merge keys, but then the `LoadFile` call needs to be replaced with creating a new object, because this is an optional feature.
I guess we should skip this for now.

### CODE REVIEW

**Comment by asdil12** [comment]
See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10107

**Comment by rwx788** [comment]
I've resolved the conflict, let's wait till CI is done and merge. Thanks again for moving this forward!

File: tools/test_yaml_valid [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 use Test::More;
-use Test::YAML::Valid qw(-Tiny);
+use YAML::PP;
 
 pass("No yaml files modified") unless @ARGV;
-yaml_file_ok($_, $_) for @ARGV;
+for my $file (@ARGV) {
+    eval { YAML::PP::LoadFile($file) };
+    if (my $err = $@) {
+        fail("$file valid");
```
Comment (b10n1k) [inline tools/test_yaml_valid:9]: do you mean invalid here?

Comment (rwx788) [reply tools/test_yaml_valid:9]: Good point, should be invalid in this case.

Comment (rwx788) [reply tools/test_yaml_valid]: Fixed now, thanks for raising.

Comment (perlpunk) [reply tools/test_yaml_valid]: Well, the test is "$file valid", and in this case it failed the test that it is valid.

"Failed test $file invalid" would sound wrong.

But if you prefer it...

**Comment by okurz** [comment]
This seems to cause problems as reported yesterday evening in chat and also https://progress.opensuse.org/issues/66253 . I am suggesting a revert for now.

**Comment by rwx788** [comment]
> This seems to cause problems as reported yesterday evening in chat and also https://progress.opensuse.org/issues/66253 . I am suggesting a revert for now.

I believe that's because `{{}}` missing quotes. Let me fix that quickly.

**Comment by rwx788** [comment]
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10152 fixes the issue, so let's merge it.

**Pull request merged**